bases/shared/instagoric-server/server.js

```diff
@@ -381,64 +388,150 @@ const addRequest = (address, request) => {
+ * @param {string} address
+ * @param {string} clientType
+ * @returns {Promise<void>}
+ */
+const provisionAddress = async (address, clientType) => {
+  let { exitCode, stderr } = await nothrow($`\
+    ${agBinary} tx swingset provision-one faucet_provision ${address} ${clientType} \
+    --broadcast-mode=block \
+    --chain-id=${chainId} \
+    --from=${FAUCET_KEYNAME} \
+    --keyring-backend=test \
+    --keyring-dir=${agoricHome} \
+    --node=http://localhost:${RPC_PORT} \
+    --yes \
+  `);
+  exitCode = exitCode ?? 1;
+
+  if (exitCode)
+    console.log(
+      `Failed to provision address "${address}" of type "${clientType}" with error message: ${stderr}`,
+    );
+};
+
+/**
+ * Send funds to `address`.
+ * It only waits for the transaction
+ * checks and doesn't wait for the
+ * transaction to actually be included
+ * in a block. The returned transaction
+ * hash can be used to get the current status
+ * of the transaction later
+ * @param {string} address
+ * @param {string} amount
+ * @returns {Promise<[number, string]>}
+ */
+const sendFunds = async (address, amount) => {
+  let { exitCode, stdout } = await nothrow($`\
+    ${agBinary} tx bank send ${FAUCET_KEYNAME} ${address} ${amount} \
+    --broadcast-mode=sync \
+    --chain-id=${chainId} \
+    --keyring-backend=test \
+    --keyring-dir=${agoricHome} \
+    --node=http://localhost:${RPC_PORT} \
+    --output=json \
+    --yes \
+  `);
+  exitCode = exitCode ?? 1;
+
+  if (exitCode) return [exitCode, ''];
+  return [exitCode, String(JSON.parse(stdout).txhash)];
+};
+
 // Faucet worker.
 const startFaucetWorker = async () => {
   console.log('Starting Faucet worker!');
 
   try {
     for await (const address of subscription) {
-      // Handle request.
-      console.log('dequeued', address);
+      console.log(`dequeued address ${address}`);
       const request = addressToRequest.get(address);
 
-      const [_a, command, clientType] = request;
+      const [response, command, clientType] = request;
       let exitCode = 1;
+      let txHash = '';
+
+      console.log(`Processing "${command}" for address "${address}"`);
+
       switch (command) {
         case 'client': {
           if (!AG0_MODE) {
-            exitCode = await $`\
-          ${agBinary} tx bank send -b block \
-  ${FAUCET_KEYNAME} ${address} \
-  ${CLIENT_AMOUNT} \
-  -y --keyring-backend test --keyring-dir=${agoricHome} \
-  --chain-id=${chainId}  --node http://localhost:26657 \
-`.exitCode;
-            if (exitCode === 0) {
-              exitCode = await $`\
-            ${agBinary} tx swingset provision-one faucet_provision ${address} ${clientType} \
-  -b block --from ${FAUCET_KEYNAME} \
-  -y --keyring-backend test --keyring-dir=${agoricHome} \
-  --chain-id=${chainId} --node http://localhost:26657 \
-`.exitCode;
-            }
+            [exitCode, txHash] = await sendFunds(address, CLIENT_AMOUNT);
+            if (!exitCode)
+              pollForProvisioning(address, clientType, txHash);
```

Comment:
quick question: shouldnt this polling also happen in our transaction status endpoints?

Reply:
Didn't exactly understand your question.
Do you mean the `/transaction-status/:txhash` endpoints?

Reply:
yes.
because im assuming we're moving all types of polling to that endpoint to avoid timeouts

Reply:
So if I am getting it right, you are asking why we are not polling for the completion of the provisioning operation?
If not, can you elaborate your question?